packages/circuits/src/initial.circom

```diff
@@ -65,8 +63,8 @@ template Initial(levels, cardCount) {
     var deckPredicateSum = 0;
     var newDeckSum = 0;
     for (var i = 0; i < 2**levels - cardCount; i++) {
-        newDeckSum += newDeckLeaves[i] * (maxDeckSize**i);
-        deckPredicateSum += (maxDeckSize**i);
+        newDeckSum += newDeckLeaves[i] * (2**i);
+        deckPredicateSum += (2**i);
```

Comment:
Is this correct? I think there's a problem where card with ID 2 at position 8 and card with ID 8 at position 2 would end up counting for the same weight in the sum, hence not constraining the deck as we want, right?

Reply:
Oh yeah, there could be collision, didn't think of that. The reason that I change it to 2 instead of using `maxDeckSize` is because of overflow issue on the smart contract side (i.e. when we do calculation of `64**64` to get the `deckPredicate`). Even if we use modulo, we still cannot run away from needing to do these calculations. What do you think could be a good way to tackle this?

Reply:
Why can't we use the modulus on the smart contract too?

But also ... do we need to make these calculations on the contract side? I think this can be private signals if we pick caids in the circuits (which we must do, cf. other comment)

Reply:
For example, if on this [line](https://github.com/norswap/0xFable/blob/initial-hand-verification/packages/contracts/src/Game.sol#L494), using a modulus would be like `pubSignals[i] = 64**deckCount % (2**256)` if I'm thinking correctly? Needing to calculate `64**deckCount` already overflows although the final value is not.

Reply:
You would need to use the exact prime that Circom uses (and not 2^256) otherwise the contract vs circuit calculation wouldn't match (but again, I don't think this needs to be computed on the contract at all).

In general, you want to use a large prime to compute these checksums, as using something that is not can lead to collisions.